apps/src/templates/hintsDisplay.jsx

```diff
@@ -0,0 +1,57 @@
+module.exports = React.createClass({
+
+  propTypes: {
+    hintReviewTitle: React.PropTypes.string.isRequired,
+    getSeenHints: React.PropTypes.func.isRequired,
+    getUnseenHints: React.PropTypes.func.isRequired,
+    recordUserViewedHint: React.PropTypes.func.isRequired,
+    lightbulbSVG: React.PropTypes.node.isRequired,
+  },
+
+  getInitialState: function () {
+    return {
+      seenHints: this.props.getSeenHints(),
+      unseenHints: this.props.getUnseenHints(),
+      showViewHintButton: true
+    };
+  },
+
+  viewHint: function () {
+    this.props.recordUserViewedHint(this.state.unseenHints[0]);
+    var newState = this.getInitialState();
```

Comment:
Yeah, I wasn't sure whether to go with that or with

```javascript
setState({
  seenHints: this.props.getSeenHints(),
  unseenHints: this.props.getUnseenHints(),
  showViewHintButton: true
});
````

Really what I want is a "refresh state" concept